consumergroup.go

```diff
@@ -169,27 +190,32 @@ func (cgi *ConsumergroupInstance) Registered() (bool, error) {
 	return cgi.cg.kz.exists(node)
 }
 
-// Register registers the consumergroup instance in Zookeeper.
-func (cgi *ConsumergroupInstance) Register(topics []string) error {
-	if exists, err := cgi.Registered(); err != nil {
-		return err
-	} else if exists {
-		return ErrInstanceAlreadyRegistered
+// Registered returns current registration of the consumer group instance.
+func (cgi *ConsumergroupInstance) Registration() (*Registration, error) {
+	node := fmt.Sprintf("%s/consumers/%s/ids/%s", cgi.cg.kz.conf.Chroot, cgi.cg.Name, cgi.ID)
+	val, _, err := cgi.cg.kz.conn.Get(node)
+	if err != nil {
+		return nil, err
 	}
 
-	// Create a JSON record that is compatible with the JVM consumer.
-	// I am not quite sure what this information is used for.
+	reg := &Registration{}
+	if err := json.Unmarshal(val, reg); err != nil {
+		return nil, err
+	}
+	return reg, nil
+}
 
+// Register registers the consumergroup instance in Zookeeper.
+func (cgi *ConsumergroupInstance) Register(topics []string) error {
 	subscription := make(map[string]int)
 	for _, topic := range topics {
 		subscription[topic] = 1
 	}
-
-	data, err := json.Marshal(map[string]interface{}{
-		"pattern":      "white_list",
-		"subscription": subscription,
-		"timestamp":    fmt.Sprintf("%d", time.Now().Unix()),
-		"version":      1,
+	data, err := json.Marshal(&Registration{
+		Pattern:      RegPatternStatic,
+		Subscription: subscription,
+		Timestamp:    time.Now().Unix(),
+		Version:      RegDefaultVersion,
 	})
```

Comment:
Should this overwrite the current subscription, or amend it?

E.g. `foo: 1 & bar: 1` + `foo: 1 & baz: 1` = `foo: 2 & bar: 1 & baz: 1`